server/tests/src/test/java/org/infinispan/server/functional/BackupManagerIT.java

```diff
@@ -419,6 +419,13 @@ private void performTest(Function<RestClient, RestResponse> backupAndDownload,
       RestResponse getResponse = backupAndDownload.apply(client);
       String fileName = getResponse.header("Content-Disposition").split("=")[1];
 
+      // Copy the returned zip bytes to the local working dir
+      File backupZip = new File(WORKING_DIR, fileName);
+      try (InputStream is = getResponse.bodyAsStream()) {
+         Files.copy(is, backupZip.toPath(), StandardCopyOption.REPLACE_EXISTING);
+      }
+      getResponse.close();
```

Comment:
Funny thing I learned with this. The JDK HTTP client only considers that a request has finished if the response is closed/the body is read. Adding more entries in the test, say 1_000, would cause the JDK client close method to hang since this last request here hasn't "finished" with the close.

I guess it needs some concurrency because, with fewer requests, it finishes OK. https://bugs.openjdk.org/browse/JDK-8308364 complete with won't fix.

Reply:
Yeah, not having AutoCloseable on this from the start was stupid